components/brave_rewards/resources/creator_detection/creator_detection.ts

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import { log, waitForURLChange, singleCall, CreatorInfo } from './helpers'
+
+import * as github from './github'
```

Comment:
Making a one universal script can become a perf bottle neck (depends on the script sizes).

Reply:
Split the scripts into separate bundles 👍